File: src/main/java/org/example/kotlinWorkshop/java/lesson02/t3_for/t3_2/ForEach.java
package org.example.kotlinWorkshop.java.lesson02.t3_for.t3_2;

import java.util.List;

public class ForEach {
  public void printAll(List<String> list) {
    for (String item: list) {
      System.out.println(item);
    }
  }

  public void printAllFunctional(List<String> list) {
    list.forEach((item) -> System.out.println(item));
  }

  public void printAllFunctional2(List<String> list) {
    list.forEach(System.out::println);
  }
}

File: src/main/kotlin/org/example/kotlinWorkshop/kotlin/lesson02/t3_for/t3_2/ForEach.kt
package org.example.kotlinWorkshop.kotlin.lesson02

class CBForEach {
    fun printAll(list: List<String?>) {
        for (item in list) {
            println(item)
        }
    }

    fun printAllFunctional(list: List<String?>) {
        list.forEach { println(it) }
    }

    fun printAllFunctional2(list: List<String?>) {
        list.forEach(::println)
    }
}